packages/webapp/src/containers/Animals/SingleAnimalView/index.tsx

```diff
@@ -98,7 +188,7 @@ function SingleAnimalView({ isCompactSideMenu, history, match }: AddAnimalsProps
         cancelModalTitle={t('ANIMALS.EDIT_ANIMAL_FLOW')}
         isEditing={isEditing}
         setIsEditing={setIsEditing}
-        key={isEditing ? 'edit' : 'readonly'}
+        key={`${JSON.stringify(defaultFormValues)}${isEditing}`}
```

Comment:
It would be great if we could avoid giving such a long key, but new values are not reflected after editing and saving when using the previous keys like 'edit' or 'readonly'. @antsgar Do you think there’s a better way to force updates without using this long key?

Reply:
Oh I missed this comment, sorry! I don't think we should be using a key to force a "reset" of the component at all in the first place. It seems if I remove this key the form breaks because it can't find the step to render next. I think we could solve this by resetting the form component by setting the active step back to 0 when the form is saved. So in `ContextForm`, I'd change what's passed to the child component as the `onSave` prop like so:

```js
<Component
      history={history}
      steps={steps}
      activeStepIndex={activeStepIndex}
     ...
      onSave={() => {
        setActiveStepIndex(0);
        onSave?.();
      }}
      {...form}
      {...props}
    >
```

and then remove the key altogether in this line.

Reply:
Ah, this part of the form has given me so much grief!! I even once did [remove the key](558f3b6d2fe84475c1103811012a4a58f0850e3d) but restored it when I saw the crash (instead of solving the crash!!), so this is so great, thank you so much 🙇 🙏 

Also remounting the form I had also been masking two other issues
- the values incorrectly being saved when editing > hit cancel > yes to modal prompt, and
- editing > hit cancel > yes to modal prompt => modal doesn't dismiss

Those just needed to be controlled explicitly so I have added those changes too. And so I _think_ the form is now now at 100% ideal behaviour!! 🤞 No flash of values and no ugly key!! 🎉